docs/blitz/design/advanced_actions/advanced.rst

```diff
@@ -0,0 +1,514 @@
+        value:         # a list of device names
+          - PE1
+          - PE2
+        parallel: True
+        actions:
+          - configure:
+              # The action name
+              alias: execute_
+              device: "%VARIABLES{devices}" 
+              command: feature bgp
+
+.. note::
+
+  In **Example-2** script would configure 2 devices with same command in parallel, without writing duplicate blocks of text.
+
+**Example-3: Executing actions until the api output is passed** 
+
+Loop over actions for maximum time of 5 seconds, execute actions once (one iteration).
+If the result of first action was not equal to "passed",
+terminate the loop, else continue until the condition is met or
+max_time is reached
+
+
+.. code-block:: YAML
+
+
+    - loop:
+        # Loop over an action at least running it once and if a condition met terminate the loop
+        do_until: "%VARIABLES{api_mtu_size} != passed"
+        max_time: 5
+        actions:
+              - api:
+                  alias: api_mtu_size
+                  description: get the api value and verify the output
+                  device: "%{testbed.devices.PE1.alias}"
+                  function: get_interface_mtu_size
+                  save:
+                    - variable_name: nbc
+                  arguments:
+                    interface: GigabitEthernet1
+              - execute:
+                  command: show vrf
+                  device: PE2
+
+**Example-4: Executing actions until the api output is passed** 
+
+Looping over an action twice (two iteration) since the range is 2, and each time,
+and run a couple of actions in parallel
+Also after each parallel call sleep for amount of the range value, so once for one second and the other for two seconds.
+
+.. code-block:: YAML
+
+
+    - loop:
+        range: 2
+        loop_variable_name: range_name
+        actions:
+          - parallel:
+            - parse:
+                device: PE1
+                command: show version
+            - execute:
+                device: PE2
+                command: show version
+        - sleep:
+            sleep_time: "%VARIABLES{range_name}"
+
+**Example-5: Synchronizing with every_seconds**
+
+The keyword ``every_seconds`` is defined so users can manage their loop and if possible run it with synchronized timing.
+If the execution of an iteration of a loop exceeds the time assigned for every_seconds, the loop would still continue its work but a warning would be
+printed into the log. **Example-5** shows how ``every_seconds`` work.
+
+This action is looping over a list of size two, hence two iteration and each iteration should take 8 seconds
+if the iteration ends in less than 8 seconds, the loop would sleep for the remaining of that time and after reaching 8 seconds
+it would execute the other iteration. The total time of execution in this case would be 16 seconds
+Keep in mind if an iteration takes more than 8 seconds the loop continue the work and it wont stop

```

Comment:
Keep in mind if an iteration takes more than 8 seconds the action will continue, it will not terminate after 8 seconds. The 8 second is the minimal interval between the loop.

Reply:
done